msteams-platform/tabs/how-to/using-teams-client-sdk.md

```diff
@@ -1,72 +1,469 @@
+
+Once you're able, the next step is to [update existing application code](#2-update-sdk-references) with the changes described in this article. In the meantime, the v.1 to v.2 API translation layer provides backwards compatibility, ensuring your existing Teams app continues to work in TeamsJS version 2.0.
+
+#### Teams apps running across Microsoft 365
+
+Enabling an existing Teams app to run in Outlook and Office requires all of the following:
+
+1. Dependency on TeamsJS version 2.0 ( `@microsoft/teams-js@2.0`) or later,
+
+2. [Modifying existing application code](#2-update-sdk-references) according to the required changes described in this article, and
+
+3. [Updating your app manifest](#3-update-the-manifest-optional) to version 1.13 or later.
+
+For more info, see [Extend Teams apps across Microsoft 365](../../m365-apps/overview.md).
+
+### Callbacks converted to promises
+
+Teams APIs that previously took a callback parameter have been updated to return a JavaScript [Promise](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) object. These include the following APIs:
+
+```js
+app.getContext, app.initialize, appInstallDialog.openAppInstallDialog, authentication.authenticate, authentication.getAuthToken, authentication.getUser, authentication.registerAuthenticationHandlers was removed to support using Promises, calendar.openCalendarItem, calendar.composeMeeting, call.startCall, chat.getChatMembers, chat.openConversation, core.executeDeepLink, files.addCloudStorageFolder, files.deleteCloudStorageFolder, files.getCloudStorageFolderContents, files.getCloudStorageFolders, files.getFileDownloads, legacy.fullTrust.getConfigSetting, legacy.fullTrust.joinedTeams.getUserJoinedTeams, location.getLocation, location.showLocation, mail.openMailItem, mail.composeMail, media.captureImage, media.getMedia, media.selectMedia, media.viewImages, media.scanBarCode, meeting.getAppContentStageSharingState, meeting.getAppContentStageSharingCapabilities, meeting.getAuthenticationTokenForAnonymousUser, meeting.getIncomingClientAudioState, meeting.getLiveStreamState, meeting.getMeetingDetails, meeting.requestStartLiveStreaming, meeting.requestStopLiveStreaming, meeting.shareAppContentToStage, meeting.stopSharingAppContentToStage, meeting.toggleIncomingClientAudio, meetingRoom.getPairedMeetingRoomInfo, meetingRoom.sendCommandToPairedMeetingRoom, pages.backStack.navigateBack, pages.navigateCrossDomain, pages.navigateToTab, pages.tabs.getMruTabInstances, pages.tabs.getTabInstances, pages.config.setConfig, pages.config.getConfig, pages.backStack.navigateBack, people.selectPeople
```

Comment:
`chat.openConversation` was renamed `conversations.openConversation`, `core.executeDeepLink` was renamed `app.openLink`, `files.*` APIs have not been converted to use Promises and all references should be removed from the list, `legacy.fullTrust.getConfigSetting` was renamed `teams.fullTrust.getConfigSetting`, `legacy.fullTrust.joinedTeams.getUserJoinedTeams` was renamed `teams.fullTrust.joinedTeams.getUserJoinedTeams`, `media.*` APIs have not been converted to use Promises and all references should be removed from the list, `meeting.*` APIs have not been converted to use Promises and all references should be removed from the list, `pages.config.getConfig` was renamed `pages.getConfig`.